### PR TITLE
add suffcient details to unauthorized errors

### PR DESCRIPTION
**Pull request by gabemontero**
# Changes

Fixes https://github.com/tektoncd/results/issues/488

/kind cleanup

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [n/a ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [n/a ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ /] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [/ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ /] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [/ ] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ /] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [/ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Provide more informative error messages when authorization via the impersonation feature fails.
```

@sayan-biswas FYI

### CODE REVIEW

File: pkg/api/server/v1alpha2/auth/rbac.go [hunk 1]
```diff
@@ -120,7 +122,8 @@ func (r *RBAC) Check(ctx context.Context, namespace, resource, verb string) erro
 		if impersonator != nil {
 			if err := impersonator.Check(ctx, r.authz, user); err != nil {
 				log.Println(err)
-				return status.Error(codes.Unauthenticated, "permission denied")
+				retMsg := fmt.Sprintf("%s: %s", retMsg, err.Error())
```
Comment (khrm) [inline pkg/api/server/v1alpha2/auth/rbac.go:125]: nit: We don't need to initialize a new variable. We can reuse, retMsg.
```suggestion
				retMsg = fmt.Sprintf("%s: %s", retMsg, err.Error())
```

Comment (gabemontero) [reply pkg/api/server/v1alpha2/auth/rbac.go]: updated

**Pull request merged**